### DISCUSSION TITLE
How to use Deno with a different version of TypeScript?

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [denoland/deno#16915] [Q&A] [answered]
I would like to start using TS 4.9, but it doesn't come with deno yet. Is there a way to use a different version of TypeScript?

**Comment by bartlomieju** [comment, marked as answer]
No, Deno embeds specific version of TypeScript and it's not possible to change it.

Deno v1.29 will ship with TypeScript 4.9 on December 14th.

**Reply by JacobWeisenburger** [reply]
thanks

**Reply by JacobWeisenburger** [reply]
is there somewhere that has a time line of things like this?

**Reply by bartlomieju** [reply]
You can see upcoming releases at https://github.com/denoland/deno/milestones?with_issues=no. Deno ships with latest available TypeScript release at the moment of Deno release. Last month TS released just a day or two after Deno release so it's a bit shifted this time.

**Reply by JacobWeisenburger** [reply]
Thanks, this is amazing
